Unable to build with geos 3.7.0

Building master with geos 3.7.0 fails

```
Building CXX object CMakeFiles/valhalla_build_admins.dir/src/mjolnir/valhalla_build_admins.cc.o
In file included from /tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:23:
/usr/include/geos/geom/Geometry.h:26:3: warning: #warning "The GEOS C++ API is unstable, please use the C API instead" [-Wcpp]
 # warning "The GEOS C++ API is unstable, please use the C API instead"
   ^~~~~~~
/usr/include/geos/geom/Geometry.h:27:3: warning: #warning "HINT: #include geos_c.h" [-Wcpp]
 # warning "HINT: #include geos_c.h"
   ^~~~~~~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc: In function 'std::vector<std::__cxx11::basic_string<char> > GetWkts(std::unique_ptr<geos::geom::Geometry>&)':
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:147:20: error: 'unique_ptr' is not a member of 'geos::geom::GeometryFactory'
   GeometryFactory::unique_ptr gf = GeometryFactory::create();
                    ^~~~~~~~~~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:164:35: error: 'gf' was not declared in this scope
       polys[totalpolys].polygon = gf->createPolygon(gf->createLinearRing(pline->getCoordinates()), 0);
                                   ^~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:230:21: error: 'gf' was not declared in this scope
       Polygon* poly(gf->createPolygon(polys[i].ring, interior.release()));
                     ^~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:236:41: error: 'gf' was not declared in this scope
     std::unique_ptr<Geometry> multipoly(gf->createMultiPolygon(polygons.release()));
                                         ^~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc: In function 'void BuildAdminFromPBF(const ptree&, const std::vector<std::__cxx11::basic_string<char> >&)':
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:426:20: error: 'unique_ptr' is not a member of 'geos::geom::GeometryFactory'
   GeometryFactory::unique_ptr gf = GeometryFactory::create();
                    ^~~~~~~~~~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:451:13: error: 'gf' was not declared in this scope
             gf->getCoordinateSequenceFactory()->create((size_t)0, (size_t)2));
             ^~
/tmp/valhalla/src/mjolnir/valhalla_build_admins.cc:473:41: error: 'gf' was not declared in this scope
         std::unique_ptr<Geometry> mline(gf->createMultiLineString(lines.release()));
                                         ^~
make[2]: *** [CMakeFiles/valhalla_build_admins.dir/build.make:63: CMakeFiles/valhalla_build_admins.dir/src/mjolnir/valhalla_build_admins.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:961: CMakeFiles/valhalla_build_admins.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

--- a/src/mjolnir/valhalla_build_admins.cc
+++ b/src/mjolnir/valhalla_build_admins.cc
@@ -144,7 +144,7 @@ std::vector<std::string> GetWkts(std::unique_ptr<Geometry>& mline) {
   std::vector<std::string> wkts;
 
 #if 3 == GEOS_VERSION_MAJOR && 6 <= GEOS_VERSION_MINOR
-  GeometryFactory::unique_ptr gf = GeometryFactory::create();
+  auto gf = GeometryFactory::create();
 #else
   std::unique_ptr<GeometryFactory> gf(new GeometryFactory());
 #endif

